liquigraph-core/src/main/java/org/liquigraph/core/io/ConnectionSupplier.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2014-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.liquigraph.core.io;
+
+import java.sql.Connection;
+import java.util.function.Supplier;
+
+public class ConnectionSupplier implements Supplier<Connection> {
```

Comment:
optional: you can even rename it to `LockedConnectionSupplier` to convey the semantics of `GraphJdbcConnector` (which should probably be renamed to `LiquigraphJdbcConnector` as the locking part is specific to Liquigraph)

Note: this change may not be necessary as `ClearChecksumRunner` does not need the supplier (see comment above)

Reply:
After reading one of your previous comments, I wonder why we use a ConnectionSupplier instead of calling GraphJdbcConnector#connect directly here.

I'll revert this change to keep the original codebase here.

Reply:
done in eee1086cc909db4c63be8da1800206232d13b6c7